### PR TITLE
Pointer focus

### PR DESCRIPTION
**Pull request by Bl4ckb0ne**
Gives focus to a window when pressed by the cursor

### CODE REVIEW

File: bspwc/input.c [hunk 1]
```diff
@@ -29,30 +29,47 @@ void handle_new_input(struct wl_listener *listener, void *data)
 	wlr_log(WLR_DEBUG, "New input device: %s (%s) %d:%d", device->name,
 			device_type(type), device->vendor, device->product);
 
+	uint32_t caps = 0;
```
Comment (kenogo) [inline bspwc/input.c]: Not sure if `caps` is the best variable name here. Would it hurt to just name it `capabilities`?

File: bspwc/output.c [hunk 2]
```diff
@@ -28,9 +27,11 @@ void output_frame_notify(struct wl_listener* listener, void* data)
 		wlr_renderer_clear(renderer, color);
 
 		render_tree(output->desktop->root);
+		wlr_output_render_software_cursors(wlr_output, NULL);
 
 		wlr_output_swap_buffers(wlr_output, NULL, NULL);
 	wlr_renderer_end(renderer);
+
```
Comment (kenogo) [inline bspwc/output.c]: Empty line :P

File: bspwc/tree.c [hunk 3]
```diff
@@ -64,7 +68,7 @@ bool insert_node(const struct server *server, struct node **root,
 		uint32_t width = (*root)->window->width;
 		uint32_t height = (*root)->window->height;
 
-		struct node* other_child = create_node();
+		struct node* other_child = create_node(output->desktop);
 		other_child->window = (*root)->window;
 		(*root)->window = NULL;
```
[the remark this answers is not in the document]
Comment (Bl4ckb0ne) [reply bspwc/tree.c]: TODO : https://github.com/Bl4ckb0ne/bspwc/issues/37

File: bspwc/input.c (hunk 1 above)
Comment (Bl4ckb0ne) [reply bspwc/input.c]: Agreed, I copy pasted the code from tinywl. It's fixed now.

File: bspwc/output.c (hunk 2 above)
Comment (Bl4ckb0ne) [reply bspwc/output.c]: Removed

**Pull request merged**